[Etc] 신고 기능 로직 구현

jinju9513 opened the pull request:
### 🚀: 개요
- 게시글 신고 로직 구현

### 🚀: 작업 내용
- 신고 모델 Report 정의
- Firestore 기반 신고 저장 Repository 구현
- 신고 중복 방지 로직 추가
- ViewModel에서 상태관리 (AsyncValue) 적용
- Provider 구성
- 관리자 페이지 추가
- 관리자 페이지에서 신고내용확인
- 관리자 게시글 신고 해제
- 관리자 신고 게시글 삭제

###화면
![Simulator Screenshot - iPhone 15 Pro Max - 2025-06-24 at 15 52 35](https://github.com/user-attachments/assets/d90f482b-4c0f-4ccc-b7de-cc34f209145a)
![Simulator Screenshot - iPhone 15 Pro Max - 2025-06-24 at 15 53 12](https://github.com/user-attachments/assets/f4736a7e-55a2-4a71-85c6-67b7b8b6d0e8)
![Simulator Screenshot - iPhone 15 Pro Max - 2025-06-24 at 15 53 18](https://github.com/user-attachments/assets/47118235-95ce-431d-a728-6b7745c72713)


### 🚀: 조정 파일
- report.dart
- report_repository.dart
- report_view_model.dart
- report_provider.dart
- report_test.dart 
- admin_report_repository.dart
- admin_report_screen.dart
- show_report_reason_dialog.dart
- post_detail_page.dart
- admin_report_view_model.dart

### 개발 결과
- Firestore에 신고 데이터 정상 저장 확인
- 중복 신고 차단 정상 작동
- 추후 UI 연결 시 버튼만 붙이면 바로 동작 가능
- 관리자 페이지에서 신고문서 확인 가능
- 관리자 페이지에서 신고 게시물 삭제 가능
- 관리자 페이지에서 게시물 신고 해제 가능
### issue
Closes #186

ghghtls committed:
feat:글작성 ui

jinju9513 committed:
[Etc] 신고 기능 로직 구현
Fixes #186 feat: 신고 기능용 Report 모델 정의 및 필드 추가

jinju9513 committed:
feat: Firebase 기반 신고 저장 로직 구현

jinju9513 committed:
feat: 신고 등록 기능 상태관리 및 예외 처리 추가

jinju9513 committed:
feat: 신고 관련 ViewModel 및 Repository provider 등록

jinju9513 committed:
feat : 테스트용 신고 버튼 UI 추가

jinju9513 committed:
[Etc] 신고 기능 로직 구현
Fixes #186
refactor : timestamp를 문자열에서 DateTime 객체로 변경

mJayxx committed:
feat: MyAccountPage 생성 및 데이터 연결

jinsung99123 commented on lib/repositories/scoial/report_repository.dart:
```diff
@@ -0,0 +1,31 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:travel_muse_app/models/scoial/report.dart';
+
+/// 신고 데이터를 저장하는 Repository 추상 클래스
```
저장 기능과 중복처리는 구현 잘 해주셨습니다.
현재 신고 후 처리 기능이 없으므로 추가 부탁드립니다.
예를 들어 자동 삭제 및 관리자를 거친 삭제 조치가 필요한데
배포시에는 관리자 검토후 삭제 진행이 자동삭제보다 매우 적합하다고 하네요
그렇다면 추가로 관리자 관리 구조가 필요하긴 합니다.

ghghtls committed:
feat:포스트 유효성 유틸 추가

ghghtls committed:
feat:포스트 ui와 상태변경

mJayxx committed:
feat: 로그인, 탈퇴 버튼 추가

mJayxx committed:
feat: AccountSettingPage에 계정관리 위젯 연결 및 로그인 상태 관련 코드 세부 조정

jinsung99123 committed:
feat: 로컬 노티피케이션 패키지 설치

jinsung99123 committed:
feat: ios 알림 환경 설정

mJayxx committed:
feat: appUser 문서 없는 로그인계정 자동 삭제 cloud function 구현

jinsung99123 committed:
fix: ios 재설정

jinsung99123 committed:
feat: 알림 helper 임시 설정

jinsung99123 committed:
feat: 메인 다트 알림 초기화 설정

jinsung99123 committed:
feat: timezone 패키지 추가

jinsung99123 committed:
feat: ios 추가 설정 및 UI 적용

mJayxx committed:
feat: custom dialog 위젯(WIP)

mJayxx committed:
delete: 임포트문 정리

ghghtls committed:
add:uuid 패키지 추가

ghghtls committed:
feat:글작성 저장 Ui+저장 로직 완료 , 수정+삭제 로직만 작성

ghghtls committed:
feat: 게시글 작성/수정 로직 통합 및 구조 개선

- submitPost 메서드를 도입해 게시글 생성/수정 로직 통합 처리
- 이미지 업로드 로직을 ViewModel로 이동하여 로직 책임 분리
- 게시글 수정 시 태그 tags 저장 누락 문제 해결
- PostDetailPage에서 조회수 증가 및 사용자 정보 조회 로직을 Repository 기반으로 리팩토링

jinju9513 committed:
[Etc] 신고 기능 로직 구현
Fixes #186 feat: 신고 게시글 및 댓글 모델 정의

jinju9513 committed:
feat: 신고된 게시글 및 댓글 불러오기 및 삭제/해제 로직 구현

jinju9513 committed:
fix : 불필요 주석 삭제

jinju9513 committed:
feat: 관리자용 신고 처리 ViewModel 구현

jinju9513 committed:
feat: 관리자 신고 ViewModel Provider 추가

jinju9513 committed:
feat: 관리자 신고 리스트 화면 및 기능 UI 구현

jinju9513 committed:
refactor: 신고 카드 UI 분리 및 삭제/해제 버튼 로직 연결

jinju9513 committed:
feat: 사용자 신고 처리 로직 구현 및 Firestore 연동

jinju9513 committed:
chore : 신고 기능 테스트용 화면 데이터 변경

ghghtls committed:
fix: 성향테스트 페이지_인디케이터_바 오버플로우 수정

mJayxx committed:
fix: 스플래시 분기, sns 로그인 취소 시 홈화면 무한 로딩, 온보딩 직후 테스트 시 무한 로딩 픽스

mJayxx committed:
fix: 로그아웃, 탈퇴 시 확실한 로그아웃 절차 추가, 탈퇴 시 성향테스트 데이터 정리

jinju9513 committed:
[Etc] 신고 기능 로직 구현
Fixes #186 feat : 문서 주석 추가

jinju9513 committed:
fix : 오타 수정

jinju9513 committed:
feat: 신고하기 기능 추가 및 신고 사유 입력 다이얼로그 연결

jinju9513 committed:
feat: 신고된 게시글 상세보기 연결

jinju9513 committed:
feat: 신고 게시글/댓글 닉네임 누락 시 탈퇴한 사용자 처리 추가, 문서 주석 추가

jinsung99123 committed:
feat: 웹뷰 패키지 설치

jinsung99123 committed:
feat: 웹뷰 ios 설정

jinsung99123 committed:
fix: 약관 모델 수정

jinsung99123 committed:
fix: 약관 모델 수정

jinsung99123 committed:
feat: 웹뷰 페이지 생성

jinsung99123 committed:
feat: 웹뷰 페이지 생성

jinju9513 committed:
Merge branch 'develop' into jinju9513/issue186

ghghtls committed:
fix:성향테스트인트로 닉네임 나오게 변경

jinsung99123 committed:
fix: mvvm 구조 url 추가 및 ui 반영

jinsung99123 committed:
fix: mvvm 구조 url 추가 및 ui 반영

jinsung99123 committed:
fix: 웹뷰 db에 설정 및 웹뷰 중복 방지 키 설정

jinsung99123 committed:
fix: 웹뷰 db에 설정 및 웹뷰 중복 방지 키 설정

jinsung99123 committed:
feat: 바텀시트 삭제(웹뷰 대체)

jinsung99123 committed:
fix: 마이페이지 약관 라우터 설정

jinsung99123 committed:
fix: 마이페이지 약관 라우터 설정

jinju9513 committed:
[Design] 일정삭제 ui 수정
Fixes #203 feat :일정 삭제 커스텀 다이얼로그 추가

jinju9513 committed:
fix : 일정삭제 스와이프 방식으로 변경, 디자인 반영

mJayxx committed:
feat: 프로필 이미지 등록 시 리사이징 구현

jinsung99123 committed:
fix: 약관 추가 및 마이페이지 라우터 연결

jinsung99123 committed:
fix: 약관 추가 및 마이페이지 라우터 연결

jinju9513 committed:
[Design] 일정삭제 ui 수정
Fixes #203 fix : 삭제 아이콘 text로 교체, text스타일 수정

mJayxx committed:
feat: 포스트 작성 시 썸네일 생성 cloud function

mJayxx committed:
feat: 프로필 업데이트 시 로딩 표시 ux

jinsung99123 commented on lib/models/scoial/reproted_comment.dart:
```diff
@@ -0,0 +1,40 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class ReportedComment {
```
파일명 오타있네요! 클래스명 말고 파일명

jinju9513 committed:
[Etc] 신고 기능 로직 구현
Fixes #186 fix :  파일 오타 수정

jinju9513 committed:
[Etc] 신고 기능 로직 구현
Fixes #186 fix :  누락된 webview_flutter 추가

jinju9513 committed:
Merge branch 'develop' into jinju9513/issue186
